### PR TITLE
adding hackerrank exercises in spring boot

### PR DESCRIPTION
**Pull request by ClaudiaHG23**

### CODE REVIEW

File: ClaudiaHernandezGarcia/springBootProject/src/main/java/com/globant/springBootProject/controller/HackerrankController.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("hackerrank")
+public class HackerrankController {
+    @Autowired
+    StringReverseService stringReverseService;
```
Comment (luisod78) [inline ClaudiaHernandezGarcia/springBootProject/src/main/java/com/globant/springBootProject/controller/HackerrankController.java:24]: no modifiers ? what about private?

File: ClaudiaHernandezGarcia/springBootProject/src/main/java/com/globant/springBootProject/controller/HackerrankController.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+    @Autowired
+    AnagramsService anagramsService;
+
+    @Autowired
+    PatternSyntaxCheckerService patternSyntaxCheckerService;
+
+    @PostMapping(path = "stringReverse", produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<ResponseDTO<StringReverseDTO>> getIsPalindrome(@RequestBody StringReverseDTO stringReverseDTO) {
```
Comment (luisod78) [inline ClaudiaHernandezGarcia/springBootProject/src/main/java/com/globant/springBootProject/controller/HackerrankController.java:33]: getIs.... ??? don't think is a good name for a method right?

File: ClaudiaHernandezGarcia/springBootProject/src/main/java/com/globant/springBootProject/service/AnagramsService.java [hunk 3]
```diff
@@ -0,0 +1,7 @@
+package com.globant.springBootProject.service;
+
+import com.globant.springBootProject.dto.AnagramsDTO;
+
+public interface AnagramsService {
+    AnagramsDTO Solution(AnagramsDTO anagramsDTO);
```
Comment (luisod78) [inline ClaudiaHernandezGarcia/springBootProject/src/main/java/com/globant/springBootProject/service/AnagramsService.java:6]: Solution??? uppercase for a method name??
https://google.github.io/styleguide/javaguide.html#s5.2.3-method-names

File: ClaudiaHernandezGarcia/springBootProject/src/main/java/com/globant/springBootProject/service/impl/AnagramsServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+import org.springframework.stereotype.Service;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@Service
+public class AnagramsServiceImpl implements AnagramsService {
+
+    @Override
+    public AnagramsDTO Solution(AnagramsDTO anagramsDTO) {
```
Comment (luisod78) [inline ClaudiaHernandezGarcia/springBootProject/src/main/java/com/globant/springBootProject/service/impl/AnagramsServiceImpl.java:14]: Solution??? uppercase for a method name??
https://google.github.io/styleguide/javaguide.html#s5.2.3-method-names

**Review by luisod78 — CHANGES_REQUESTED** [review]

File: ClaudiaHernandezGarcia/springBootProject/src/main/java/com/globant/springBootProject/controller/HackerrankController.java (hunk 2 above)
Comment (lrod-glb) [reply ClaudiaHernandezGarcia/springBootProject/src/main/java/com/globant/springBootProject/controller/HackerrankController.java:33]: method names still getIsSomething.... not a good name

**Review by lrod-glb — CHANGES_REQUESTED** [review]

**Review by lrod-glb — APPROVED** [review]

**Pull request merged**